packages/runtime/container-runtime/src/opLifecycle/opContentsSerializer.ts

```diff
@@ -0,0 +1,78 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	IFluidHandleContext,
+	type IFluidHandleInternal,
+} from "@fluidframework/core-interfaces/internal";
+import { assert } from "@fluidframework/core-utils/internal";
+import { FluidSerializerBase } from "@fluidframework/runtime-utils/internal";
+import { UsageError } from "@fluidframework/telemetry-utils/internal";
+
+// For passing to base class encode/stringify methods.
+const dummyBind: Partial<IFluidHandleInternal> = {
+	bind() {
+		// noop
+	},
+};
+
+/**
+ * For serializing op contents during submit. Will encode handles before serializing.
+ *
+ * @internal
+ */
+export class OpContentsSerializer extends FluidSerializerBase {
+	/**
+	 * @param root - ContainerRuntime's root handle routing context.
+	 */
+	public constructor(root: IFluidHandleContext) {
+		assert(root.routeContext === undefined, "Context provided should have been the root");
+		super(root);
+	}
+
+	/**
+	 * Encode the handles in the given op contents.
+	 * Note that the bind param is ommitted, as we don't bind handles as part of encode anymore.
+	 */
+	public encode(opContents: unknown): unknown {
+		return super.encode(opContents, dummyBind as IFluidHandleInternal);
+	}
+
+	/**
+	 * See {@link FluidSerializerBase.decode}
+	 *
+	 * @deprecated Not supported at this time
+	 */
+	public decode(_input: unknown): unknown {
```

Comment:
i don't think this is really necessary. i'd move away from inheritance and classes altogether, and just use function that can be composed and shared where necessary.

Reply:
Never mind :)